app/controllers/api/articles_controller.rb

```diff
@@ -1,31 +1,47 @@
 class Api::ArticlesController < ApplicationController
   def index
-    articles = Article.all
+    articles = category_param_check(params[:category_name])
+
     if articles.any?
       render json: { articles: articles }
     else
       render json: { message: 'There are no articles in the database' }, status: 404
     end
   end
-
+  
   def create
     article = Article.create(article_params)
+    article.category_id = set_category_id(article)
 
     if article.valid?
       render json: { message: "You have successfully added #{article.title} to the site" }, status: 201
     else
       render json: { errors: article.errors.full_messages.to_sentence }, status: 422
     end
   end
-
+  
   def show
     article = Article.find(params[:id])
     render json: { article: article }
   end
-
+  
   private
+  
+  def category_param_check(category_name)
+    if Category.pluck(:name).include? category_name
+      Article.where(category_name: params[:category_name])
+    else
+      Article.all
+    end
+  end
 
   def article_params
-    params.require(:article).permit(:title, :lede, :body)
+    params.require(:article).permit(:title, :lede, :body, :category_name, :category_id)
+  end
+
+  def set_category_id(article)
+    category_name_from_article = article.category_name
+    category_class_instance_from_categories = Category.find_by(name: category_name_from_article)
+    category_class_instance_from_categories ? category_class_instance_from_categories.id : nil
```

Comment:
It is ok, but while you change other things you can also refactor that. Try this:
`article.category_id = Category.find_by(name: article.category_name).id`
Now it actually sets category, which corresponds to naming you choose. Because in your implementation it returns of finds category, not sets it.

Reply:
We needed to keep the possibility of a null value for `category_id` so that the `article.valid?` would still trigger the error handler in case an article without an allowed category name was sent in. Our code formatter corrected(?) our ternary into the following:
```ruby
article.category_id = if Category.find_by(name: article.category_name)
                            Category.find_by(name: article.category_name).id
                          end
```